shared/src/store/action-reducers/utils/calculate-treatments.ts

```diff
@@ -1,255 +1,333 @@
-            ?.sort(
-                ({ distance: distanceA }, { distance: distanceB }) =>
-                    distanceA - distanceB
-            )
-            .map(({ patient }) => patient) ?? [];
+    // Patients that got already treated (or tried to be treated) in overrideTreatmentRange
+    // should not be treated (or tried to be treated) again in treatmentRange
+    const cateredForPatients = new Set<UUID>();
 
-    const greenPatients =
-        distancesByStatus.green
-            ?.sort(
-                ({ distance: distanceA }, { distance: distanceB }) =>
-                    distanceA - distanceB
-            )
-            .map(({ patient }) => patient) ?? [];
-
-    for (const patient of redPatients) {
-        if (
-            !caterFor(
-                catering,
+    if (cateringElement.overrideTreatmentRange > 0) {
+        const patientIdsInOverrideRange = SpatialTree.findAllElementsInCircle(
+            state.spatialTrees.patients,
+            cateringElement.position,
+            cateringElement.overrideTreatmentRange
+        );
+        // In the overrideTreatmentRange (the override circle) only the distance to the patient is important - his/her injuries are ignored
+        for (const patientId of patientIdsInOverrideRange) {
+            tryToCaterFor(
+                cateringElement,
                 catersFor,
-                patient,
-                pretriageEnabled,
-                bluePatientsEnabled
-            )
-        ) {
-            break;
+                getElement(state, 'patients', patientId),
+                state.configuration.pretriageEnabled,
+                state.configuration.bluePatientsEnabled
+            );
+            cateredForPatients.add(patientId);
         }
     }
 
-    for (const patient of yellowPatients) {
-        if (
-            !caterFor(
-                catering,
-                catersFor,
-                patient,
-                pretriageEnabled,
-                bluePatientsEnabled
-            )
-        ) {
-            break;
-        }
+    if (
+        cateringElement.treatmentRange <= 0 ||
+        // Only look for more patients to treat, if there still is capacity (performance improvement)
+        (!couldCaterFor('red', cateringElement, catersFor) &&
+            !couldCaterFor('yellow', cateringElement, catersFor) &&
+            !couldCaterFor('green', cateringElement, catersFor))
+    ) {
+        return;
     }
 
-    for (const patient of greenPatients) {
-        if (
-            !caterFor(
-                catering,
-                catersFor,
+    const patientsInTreatmentRange: Mutable<Patient>[] =
+        SpatialTree.findAllElementsInCircle(
+            state.spatialTrees.patients,
+            cateringElement.position,
+            cateringElement.treatmentRange
+        )
+            // Filter out every patient in the overrideTreatmentRange
+            .filter((patientId) => !cateredForPatients.has(patientId))
```

Comment:
You can filter by `status` here as you're ignoring every non `'red' | 'yellow' | 'green'` patient later on.

Reply:
Right now `blue` patients (when `blue` patients are deactivated) are treated as `red` (else as `blue`) and `white` are treated as `yellow`, therefore `Patient.getVisibleStatus` is needed to be called before filtering patients by their status category/color.

Reply:
Right. You could filter after the `map`, though.

Reply:
Do you want directly to use `getCateringStatus` and `Patient.getVisibleStatus` after the map and groupBy directly?
Else wouldn't this result in calling multiple times the two mentioned functions?
(Or how would you calculate `patientsPerStatus` or do the loops?)

Reply:
I think this would be premature optimization. Keep in mind that we are only looking at the patients between the `overrideTreatmentRange` and `treatmentRange` around the `cateringElement`.
Additionally, we would have to iterate over all elements again to filter them and would only save the few calculations needed in `groupBy` and save a tiny bit of memory.

Reply:
> I think this would be premature optimization. Keep in mind that we are only looking at the patients between the `overrideTreatmentRange` and `treatmentRange` around the `cateringElement`. Additionally, we would have to iterate over all elements again to filter them and would only save the few calculations needed in `groupBy` and save a tiny bit of memory.

I would also leave it as it is, if you @ClFeSc have any suggestion about how to write it in an easier way, resulting in the same outcome, please make a suggestion.
Otherwise mark this as resolved if you are agreeing that this optimization is not necessarly needed.

Reply:
Yes, I agree. I forgot the fact that only patients nearby are included.